### PR TITLE
Cleanup algorithms code

### PR DESCRIPTION
**Pull request by milseman**
Delete lots of dead code, get off of collections and onto iterators, simplify implementations.

### CODE REVIEW

File: Sources/_StringProcessing/Algorithms/Algorithms/Ranges.swift [hunk 1]
```diff
@@ -11,107 +11,33 @@
...
-extension RangesCollection.Index: Comparable {
-  static func == (lhs: Self, rhs: Self) -> Bool {
-    switch (lhs.range, rhs.range) {
-    case (nil, nil):
-      return true
-    case (nil, _?), (_?, nil):
-      return false
-    case (let lhs?, let rhs?):
-      return lhs.lowerBound == rhs.lowerBound
+    public mutating func next() -> Range<Searcher.Searched.Index>? {
```
Comment (natecook1000) [inline Sources/_StringProcessing/Algorithms/Algorithms/Ranges.swift:32]: I don't know why it lets us mark these `public`…

```suggestion
    mutating func next() -> Range<Searcher.Searched.Index>? {
```

File: Sources/_StringProcessing/Algorithms/Algorithms/Ranges.swift [hunk 2]
```diff
@@ -11,107 +11,33 @@
...
-    switch (lhs.range, rhs.range) {
-    case (nil, _):
-      return false
-    case (_, nil):
-      return true
-    case (let lhs?, let rhs?):
-      return lhs.lowerBound < rhs.lowerBound
-    }
+extension RangesSequence: Sequence {
+  public func makeIterator() -> Iterator {
```
Comment (natecook1000) [inline Sources/_StringProcessing/Algorithms/Algorithms/Ranges.swift:39]:
```suggestion
  func makeIterator() -> Iterator {
```

File: Sources/_StringProcessing/Algorithms/Algorithms/Split.swift [hunk 3]
```diff
@@ -126,12 +126,12 @@ extension Collection {
 
 extension Collection where Element: Equatable {
   @_disfavoredOverload
-  func split<C: Collection>(
+  func _split<C: Collection>(
```
Comment (natecook1000) [inline Sources/_StringProcessing/Algorithms/Algorithms/Split.swift:129]: Nice 👍🏻

**Comment by milseman** [comment]
@swift-ci please test

**Review by natecook1000 — COMMENTED** [review]
This is great, @milseman! 👏🏻

Only nit is the `public` declarations on internal types.

File: Sources/_StringProcessing/Algorithms/Algorithms/Split.swift [hunk 4]
```diff
@@ -11,15 +11,15 @@
 
 // MARK: `SplitCollection`
 
-struct SplitCollection<Searcher: CollectionSearcher> {
-  public typealias Base = Searcher.Searched
+struct SplitSequence<Searcher: CollectionSearcher> {
+  public typealias Input = Searcher.Searched
```
Comment (milseman) [inline Sources/_StringProcessing/Algorithms/Algorithms/Split.swift:15]:
```suggestion
  typealias Input = Searcher.Searched
```

File: Sources/_StringProcessing/Algorithms/Algorithms/Split.swift [hunk 5]
```diff
@@ -29,53 +29,53 @@ struct SplitCollection<Searcher: CollectionSearcher> {
...
-    self.ranges = base._ranges(of: searcher)
+    self.ranges = input._ranges(of: searcher)
     self.maxSplits = maxSplits
     self.omittingEmptySubsequences = omittingEmptySubsequences
   }
 }
 
-extension SplitCollection: Sequence {
+extension SplitSequence: Sequence {
   public struct Iterator: IteratorProtocol {
```
Comment (milseman) [inline Sources/_StringProcessing/Algorithms/Algorithms/Split.swift:44]:
```suggestion
  struct Iterator: IteratorProtocol {
```

**Comment by milseman** [comment]
@swift-ci please test

**Review by natecook1000 — APPROVED** [review]

**Pull request merged**